### PR TITLE
[9.0][website_canonical_url] Migrate.

### PR DESCRIPTION
**Pull request by yajo**
Did almost nothing from 8.0 branch. Just a squash on Transifex's commits, and most basic changes.

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
What you can do is to propose this to master.

**Comment by yajo** [comment]
I'd love to, but I think I'd need authorisation from @tremlin and @rami-wafaie, because that would imply a license change.

**Comment by pedrobaeza** [comment]
OK, let's see what they say, but we can also rewrite the code from scratch (and we'll probably need it as v10 controller handling is different - even there are already included some things I think - like the pager).

**Comment by pedrobaeza** [comment]
@Yajo, tests are not correct.

**Comment by yajo** [comment]
https://github.com/OCA/server-tools/pull/666 should have fixed it, I rebased, let's see.

**Comment by yajo** [comment]
Finally [tests are working](https://travis-ci.org/OCA/website/jobs/187151583#L1013-L1030) 😅 

However Travis went ❌ because of some weird thing... [Only OCB build fails with `website_portal_contact` tour, with no traceback](https://travis-ci.org/OCA/website/jobs/187151583#L1347-L1374)... It makes no sense, Odoo build works!

I think I need some help with that...

**Comment by pedrobaeza** [comment]
I have no clue either. Have you checked commit history to see if there's an specific commit for that module in OCB that it's not in Odoo?

**Comment by yajo** [comment]
But it is an OCA addon.

**Comment by pedrobaeza** [comment]
Uhm, and it's failing also on main branch?

**Comment by yajo** [comment]
Seems like not.

**Comment by pedrobaeza** [comment]
Yeah, I see that it's failing in both Odoo and OCB. First step should be to fix the main branch.

**Comment by yajo** [comment]
There was a problem with Sass indentation and encoding I fixed, please check again now, only OCB fails.

**Comment by yajo** [comment]
Yay! #306 made Travis 💚

**Comment by pedrobaeza** [comment]
Please squash a bit commits from 8.0

**Comment by yajo** [comment]
Done.

**Review by pedrobaeza — APPROVED** [review]

**Review by cubells — APPROVED** [review]
:+1: 

LGTM

**Comment by rruebner** [comment]
@Yajo We want to use this add-on for odoo 10. Is there a PR for odoo 10 already in progress? Otherwise we would migrate it to odoo 10.

File: website_canonical_url/models/website.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                url_parts[2] = lang + url_parts[2]
+                canonical_url = urlunparse(url_parts)
+        # Special case for rerouted requests to root path
+        try:
+            if (canonical_url.startswith("/page/") and
+                    self.menu_id.child_id[0].url == canonical_url):
+                canonical_url = "/"
+        except:  # pragma: no cover
+            pass
+        return canonical_url
```
Comment (rruebner) [inline website_canonical_url/models/website.py:35]: @Yajo Do you think it is enough to use relative paths as canonical links? Based on the [google docs](https://support.google.com/webmasters/answer/139066?visit_id=1-636293281057402117-902391869&hl=en&rd=1) you should avoid using relative urls.

Comment (yajo) [reply website_canonical_url/models/website.py:35]: I did not know about that, definitely a 💩, but it should be fixed.

**Comment by yajo** [comment]
@rruebner you can check https://github.com/OCA/website/issues/262 to see migration attempts for v10. Normally the first thing you have to do when you want to migrate something is to go there and ask a maintainer to add you as the migrator for X addon, and when it's done ask them to link the PR. It's or way to avoid dupe efforts.

**Pull request merged**